app/models/charge.rb

```diff
@@ -57,4 +57,10 @@ def validate_percentage
 
     validation_result.error.each { |error| errors.add(:properties, error) }
   end
+
+  def validate_currency
+    return if plan.amount_currency == amount_currency
```

Comment:
Why not directly setting the currency of the charge based on the currency of the plan in that case?

Not sure to understand what we're trying to solve here.. 🤔

Reply:
the currency can be set in the payload, we may change this if we want to force the currency of the plan, but it's out of the bug scope for me